BrowserKit/Tests/SiteImageViewTests/ImageProcessing/SiteImageCacheTests.swift

```diff
@@ -0,0 +1,132 @@
+            XCTAssertEqual("Unable to retrieve image from cache with reason: The request is empty or `nil`.",
+                           error.description)
+        } catch {
+            XCTFail("Should have failed with ImageError type")
+        }
+    }
+
+    func testGetFromCache_whenEmptyImage_returnsError() async {
+        let subject = DefaultSiteImageCache(imageCache: imageCache)
+
+        do {
+            let _ = try await subject.getImageFromCache(domain: "www.example.com", type: .heroImage)
+
+        } catch let error as ImageError {
+            XCTAssertEqual(imageCache.capturedRetrievalKey, "www.example.com-heroImage")
+            XCTAssertEqual("Unable to retrieve image from cache with reason: Image was nil",
+                           error.description)
+        } catch {
+            XCTFail("Should have failed with ImageError type")
+        }
+    }
+
+    func testGetFromCache_whenImage_returnsSuccess() async {
+        let expectedImage = UIImage()
+        imageCache.image = expectedImage
+        let subject = DefaultSiteImageCache(imageCache: imageCache)
+
+        do {
+            let result = try await subject.getImageFromCache(domain: "www.example2.com", type: .favicon)
+            XCTAssertEqual(imageCache.capturedRetrievalKey, "www.example2.com-favicon")
+            XCTAssertEqual(expectedImage, result)
+
+        } catch {
+            XCTFail("Should have succeeded with image")
+        }
+    }
+
+    // MARK: - Cache image
+
+    func testCacheImage_whenError_returnsError() async {
+        let expectedImage = UIImage()
+        imageCache.storageError = KingfisherError.requestError(reason: .emptyRequest)
+        let subject = DefaultSiteImageCache(imageCache: imageCache)
+
+        do {
+            let _ = try await subject.cacheImage(image: expectedImage, domain: "www.firefox.com", type: .favicon)
+
+        } catch let error as ImageError {
+            XCTAssertEqual(imageCache.capturedStorageKey, "www.firefox.com-favicon")
+            XCTAssertEqual(imageCache.capturedImage, expectedImage)
+            XCTAssertEqual("Unable to cache image with reason: The request is empty or `nil`.",
+                           error.description)
+        } catch {
+            XCTFail("Should have failed with ImageError type")
+        }
+    }
+
+    func testCacheImage_whenSuccess_returnsSuccess() async {
+        let expectedImage = UIImage()
+        let subject = DefaultSiteImageCache(imageCache: imageCache)
+
+        do {
+            let _ = try await subject.cacheImage(image: expectedImage, domain: "www.firefox.com", type: .favicon)
+            XCTAssertEqual(imageCache.capturedStorageKey, "www.firefox.com-favicon")
+            XCTAssertEqual(imageCache.capturedImage, expectedImage)
+
+        } catch {
+            XCTFail("Should have succeeded")
+        }
+    }
+}
+
+private class MockDefaultSiteImageCache: DefaultImageCache {
```

Comment:
nit: The name is confusing because it's a mix of the SiteImageCache and DefaultImageCache. Should be MockDefaultImageCache.